### ISSUE TITLE
Page crashes when switching/dragging nodes around

### ISSUE THREAD

**Issue by kevinschaich** [xyflow/xyflow#4728]
### What platform were you using when you found the bug?

- React Flow / Svelte Flow version: 12.3.2
- Browser and version: Arc Version 1.63.1 (54714)
- OS and version: MacOS Sequoia 15.1

### Describe the Bug

The last few hours in our app there's a nasty bug that we haven't been able to track down. Sometimes it throws an error in the console and sometimes it does not. I managed to catch it on video. 

`Can't read property of undefined - reading 'internals'`

<img width="1589" alt="image" src="https://github.com/user-attachments/assets/1e1937aa-02b7-4ffa-8134-b1cfab381803">

<img width="1924" alt="image" src="https://github.com/user-attachments/assets/54bf647b-00ff-4b5b-a01b-17a8448ebec4">

It's failing here:

https://github.com/xyflow/xyflow/blob/de1098ad31b00e2ac69c96f942ff4a10198313ab/packages/system/src/xydrag/utils.ts#L77-L119

Specifically this line:

https://github.com/xyflow/xyflow/blob/de1098ad31b00e2ac69c96f942ff4a10198313ab/packages/system/src/xydrag/utils.ts#L109

^ There's a TypeScript non-null assertion for that object that appears to be incorrect in some edge case.

### Steps to reproduce the bug or issue

1. Quickly click around, change node selections, drag nodes
2. Page crashes

### Expected behavior

Page does not crash

### Screenshots or Videos

Ignore the cursor not moving – i think my screen recorder is attempting to smooth out the motion so it's not rapid back and forth movement

https://github.com/user-attachments/assets/d312a854-63fc-4170-bfb0-0224da07a754

**Comment by bcakmakoglu** [comment]
Instead of a video could you please provide a proper reproduction of the problem?

You can find a template right here: https://new.reactflow.dev/react-ts

**Comment by kevinschaich** [comment]
@bcakmakoglu Unfortunately I can't share our source code, and a minimal reproduction is likely not to hit the bug (it would be very similar to the default sandbox). Hoping we can find a way to make progress on this nonetheless.

I feel like a video repro and the exact line number where things are failing should be sufficient to debug further – I think that non-null type assertion is getting hit and we could try adding some alternative logic in if `nodeLookup.get(nodeId)` is actually undefined.

**Comment by peterkogo** [comment]
Having such a loose non-null assertion without a reason like that is terrible! Thanks for bringing it up, I will fix this.

However! What exactly am I seeing in the video? It does not really look like you are dragging the nodes. 
Furthermore, there should be no real reason for the lookup of the node failing. The node wouldn't even be rendered if it was not in the `nodelookup`.

**Comment by jamal-khey** [comment]
@peterkogo i have a similar issue, but only when i have the minimap open:

[crash.webm](https://github.com/user-attachments/assets/5d73d6a9-c064-40e8-8f50-e7a45c633ffc)
[working.webm](https://github.com/user-attachments/assets/85da3614-c995-43bc-95e6-0ea03c1c6fe4)

i can share my code , but i didn't succeed to reproduce the problem outside of my project. 

![image](https://github.com/user-attachments/assets/867574b8-f404-4f2f-a39e-6844516048aa)
```
function NodeComponentWrapperInner({ id: id2, nodeColorFunc, nodeStrokeColorFunc, nodeClassNameFunc, nodeBorderRadius, nodeStrokeWidth, shapeRendering, NodeComponent, onClick }) {
  const { node, x, y, width, height } = useStore((s) => {
    const node2 = s.nodeLookup.get(id2);
    const { x: x2, y: y2 } = node2.internals.positionAbsolute;
    const { width: width2, height: height2 } = getNodeDimensions(node2);`

```
`TypeError: Cannot read properties of undefined (reading 'internals')`

i can share my code [pull request Migration to v12](https://github.com/gridsuite/gridstudy-app/pull/2302) but i didn't succeed to reproduce it outside of my project.

**Comment by kevinschaich** [comment]
Thanks all, really appreciate the quick turnaround on this. Let us know if it'd be possible to get a new release out to grab this change.

We'd also be interested in sponsoring the development of the project with a pro subscription. I can sign up on the website or feel free to shoot me a note if it's easier (kevin@coblocks.ai).

**Comment by peterkogo** [comment]
@kevinschaich Thanks, greatly appreciated! Creating the pro subscription on the site is the easiest for us, but if you need a quote, or you'd like to sponsor us in some other way, you can reach out to us at info@xyflow.com

**Comment by peterkogo** [comment]
This is fixed in 12.3.3
